cluster-bootstrap/src/main/scala/akka/management/cluster/bootstrap/internal/BootstrapCoordinator.scala

```diff
@@ -114,34 +121,55 @@ private[akka] final class BootstrapCoordinator(discovery: SimpleServiceDiscovery
   private var seedNodesObservations: Map[ResolvedTarget, SeedNodesObservation] = Map.empty
 
   private var decisionInProgress = false
+  def startPeriodicDecisionTimer(): Unit =
+    timers.startPeriodicTimer(DecideTimerKey, DecideTick, settings.contactPoint.probeInterval)
+
+  private var currentDiscoveryInterval: Duration = settings.contactPointDiscovery.interval
+  def resetDiscoveryInterval(): Unit =
+    currentDiscoveryInterval = settings.contactPointDiscovery.interval
+  def backoffDiscoveryInterval(): Unit =
+    currentDiscoveryInterval = settings.contactPointDiscovery.backoffInterval(currentDiscoveryInterval)
+  def startSingleDiscoveryTimer(): Unit =
+    currentDiscoveryInterval match {
+      case d: FiniteDuration => timers.startSingleTimer(DiscoverTimerKey, DiscoverTick, d)
+      case _ => // ignore, infinite interval
+    }
 
-  timers.startPeriodicTimer(ResolveTimerKey, ResolveTick, settings.contactPointDiscovery.interval)
-  timers.startPeriodicTimer(DecideTimerKey, DecideTick, settings.contactPoint.probeInterval)
+  override def preStart(): Unit = {
+    startSingleDiscoveryTimer()
+    startPeriodicDecisionTimer()
+  }
 
   /** Awaiting initial signal to start the bootstrap process */
   override def receive: Receive = {
     case InitiateBootstrapping ⇒
       log.info("Locating service members. Using discovery [{}], join decider [{}]", discovery.getClass.getName,
         joinDecider.getClass.getName)
-      lookup()
+      discoverContactPoints()
 
       context become bootstrapping(sender())
   }
 
   /** In process of searching for seed-nodes */
   def bootstrapping(replyTo: ActorRef): Receive = {
-    case ResolveTick ⇒
-      lookup()
+    case DiscoverTick ⇒
+      discoverContactPoints()
+    // the next round of discovery will be performed once this one returns
```

Comment:
The (scalariform?) indent suggests that this comment belongs to the next case - perhaps move it up a line?

Reply:
yeah weird... will check